### PR TITLE
Podcasting: Update My Home to display launched podcast site message

### PR DESCRIPTION
**Pull request by allilevine**
#### Changes proposed in this Pull Request

* Display "You launched your podcast site!" title and description for "Site created" step when the podcasting site has been launched.

#### Testing instructions

* Create a podcast site, but don't launch it.
* With this branch checked out, go to "My Home" for the site you created (http://calypso.localhost:3000/home/your-podcast-site-slug.wordpress.com)
* Click on the "Site Created" step under Site Setup if a different step is visible. You should see the podcast welcome message:
Screen Shot 2021-02-12 at 4 22 01 PM

* Launch the site (e.g. from the editor or Edit Homepage).
* When you land on "My Home" again, click "Show Site Setup" and then on the "Site Created" step under Site Setup if a different step is visible. You should see the podcast launched message:
Screen Shot 2021-02-12 at 4 25 33 PM

Related to 437-gh-Automattic/dotcom-manage

### CODE REVIEW

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js [hunk 1]
```diff
@@ -83,8 +84,15 @@ export const getTask = (
 
 			// Change the task title for podcasting sites.
 			if ( isPodcastingSite ) {
-				taskData.title = translate( 'Welcome to your podcast site!' );
+				taskData.title = isLaunchedSite
+					? translate( 'You launched your podcast site!' )
```
Comment (a8ci18n) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js:88]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'You launched your site!' )` *ES Score: 7*

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js [hunk 2]
```diff
@@ -83,8 +84,15 @@ export const getTask = (
...
+					: translate( 'Welcome to your podcast site!' );
 				taskData.hideLabel = true;
+				taskData.description = isLaunchedSite
+					? translate(
+							'Now that you’ve created your site, we’ll guide you through completing a few additional steps to continue building your site.'
+					  )
+					: translate( "Next, we'll guide you through setting up and launching your site." );
```
Comment (kwight) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js]: Can this use single quotes and an apostrophe like the line above for consistency?

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js [hunk 3]
```diff
@@ -83,8 +84,15 @@ export const getTask = (
...
 			// Change the task title for podcasting sites.
 			if ( isPodcastingSite ) {
-				taskData.title = translate( 'Welcome to your podcast site!' );
+				taskData.title = isLaunchedSite
+					? translate( 'You launched your podcast site!' )
+					: translate( 'Welcome to your podcast site!' );
 				taskData.hideLabel = true;
+				taskData.description = isLaunchedSite
+					? translate(
+							'Now that you’ve created your site, we’ll guide you through completing a few additional steps to continue building your site.'
```
Comment (kwight) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js:93]: Can this be more concise and closer to the un-launched version, maybe "Next, we'll guide you through completing your site setup."

**Comment by kwight** [comment]
This works well, but when paired with the necessary D56858-code, I see the same as @allilevine : after launching the site, we get immediately shown the next incomplete step (email confirmation), rather than being able to see the update content on the first step.

@allilevine I feel like there's a way to either defeat the auto-advance, or explicitly set what step to land on, I can't remember. Digging a little bit, maybe you could look at `advanceToNextIncompleteTask` around the `SiteSetupList` component, and see if there's a way to remain on the first step (where we see that messaging).

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js (hunk 3 above)
Comment (allilevine) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js:93]: That makes me wonder, do we need two different messages here? Can we say "Welcome to your podcast site!" regardless of whether it's launched? And use "Next, we'll guide you through completing your site setup." for both?

cc @sfougnier

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js [hunk 4]
```diff
@@ -83,8 +84,15 @@ export const getTask = (
...
+					  )
+					: translate( 'Next, we’ll guide you through setting up and launching your site.' );
```
Comment (a8ci18n) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js:95]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Next, we'll guide you through setting up and launching your site.' )` *ES Score: 7*

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js (hunk 3 above)
Comment (kwight) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js:93]: While it's nice to celebrate the actual launch, it's also nice to have the simplicity – I'm good with whatever @sfougnier prefers.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js (hunk 2 above)
Comment (allilevine) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js]: Fixed in https://github.com/Automattic/wp-calypso/pull/50056/commits/6ebdc298bed34d99c40f7d60a64245690aec8dab

**Comment by allilevine** [comment]
> This works well, but when paired with the necessary D56858-code, I see the same as @allilevine : after launching the site, we get immediately shown the next incomplete step (email confirmation), rather than being able to see the update content on the first step.
> 
> @allilevine I feel like there's a way to either defeat the auto-advance, or explicitly set what step to land on, I can't remember. Digging a little bit, maybe you could look at `advanceToNextIncompleteTask` around the `SiteSetupList` component, and see if there's a way to remain on the first step (where we see that messaging).

@kwight Thanks for the pointer! I set `completeOnView` to `false` for the first step, but strangely I'm also seeing the update content on launch again.

Here's a video of the flow with this change and the patch applied. To get around the WordPress.com redirect I quickly closed the tab and opened Calypso dev: https://www.loom.com/share/acd095c6e661402ba6e1ad29e60a48d8

Here's the flow with just the patch: https://www.loom.com/share/5ca286cffa624787869cc21c90bc14ab

**Comment by kwight** [comment]
I deployed D56858-code to more easily play with all of this, and I have a proposal.

I don't see a problem with showing the "You launched your site!" celebration card (in fact, I quite like it). I admit "I set completeOnView to false for the first step" doesn't feel right, and means more complicated code (and the ternaries for different "launch vs. not" text are already awkward).

How about we:
* keep the launch celebration card,
* keep the "Welcome to your podcast site!" title for all cases,
* leave the description text as-is.

This will make all card content make sense regardless of seeing the welcome card before or after launch, and we don't need to add special handling to the checklist for launch (advancing to the next incomplete task is no longer a problem, because users will still see a launch celebration card on launch). I know @creativecoder voiced a concern in the original issue of "given that Anchor users will be launching a site without customizing the home page, I think it's important we lead them to this checklist (which includes updating the home page) as directly as possible", but now our flow drops the user into homepage editing right away, rather than a post – so that feel like less of a problem now. It also brings the whole Anchor experience more inline with ordinary site experiences in Calypso.

To make the above happen we would not need this PR at all, and would just need to add back the card on the back-end (which I've done in D57286-code).

Is this a good solution at least for launch @sfougnier ?

**Comment by allilevine** [comment]
> 
> How about we:
> 
>     * keep the launch celebration card,
> 
>     * keep the "Welcome to your podcast site!" title for all cases,
> 
>     * leave the description text as-is.
> 
> 
> This will make all card content make sense regardless of seeing the welcome card before or after launch, and we don't need to add special handling to the checklist for launch (advancing to the next incomplete task is no longer a problem, because users will still see a launch celebration card on launch). 

Just to confirm with a visual 😅 this means we'll see this welcome card before launch, and after launch it will display after the launch celebration card:
Screen Shot 2021-02-18 at 2 20 28 PM

**Comment by kwight** [comment]
> we'll see this welcome card before launch

Correct!

> and after launch it will display after the launch celebration card

Correct – but the user will not see the specific "site created" card in your screenshot, because after the celebratory launch card, My Home will load the checklist with the next unfinished item displayed (the auto-advance we were fighting against). If they click on the "site created" step themselves, they'll see the screengrab you've got (just the same as ordinary WP.com users).

**Comment by ghost** [comment]
> Is this a good solution at least for launch?

I just confirmed that this indeed matches the current behavior, so I think it makes sense to keep for launch.

**Comment by allilevine** [comment]
> I just confirmed that this indeed matches the current behavior, so I think it makes sense to keep for launch.

Thanks! Closing this PR in favor of D57286-code.

**Pull request closed without merging**